spec/datadog/ci/configuration/settings_spec.rb

```diff
@@ -53,6 +96,180 @@
+          around do |example|
+            ClimateControl.modify(Datadog::CI::Ext::Settings::ENV_AGENTLESS_URL => agentless_url) do
+              example.run
+            end
+          end
+
+          context "is not defined" do
+            let(:agentless_url) { nil }
+
+            it { is_expected.to be nil }
+          end
+
+          context "is set to some value" do
+            let(:agentless_url) { "example.com" }
+
+            it { is_expected.to eq agentless_url }
+          end
+        end
+      end
+
+      describe "#agentless_url=" do
+        it "updates the #enabled setting" do
+          expect { settings.ci.agentless_url = "example.com" }
+            .to change { settings.ci.agentless_url }
+            .from(nil)
+            .to("example.com")
+        end
+      end
+
+      describe "#instrument" do
+        let(:integration_name) { :fake }
+
+        let(:integration) { FakeIntegration.new }
+        let(:enabled) { true }
+
+        subject(:instrument) { settings.ci.instrument(integration_name, enabled: enabled) }
+
+        before do
+          settings.ci.enabled = ci_enabled
+        end
+
+        after do
+          FakeIntegration::Patcher.reset
+        end
+
+        context "ci enabled" do
+          let(:ci_enabled) { true }
+
+          context "when integration exists" do
+            it "patches the integration" do
+              expect(FakeIntegration::Patcher).to receive(:patch)
+
+              instrument
+            end
+
+            context "when called multiple times" do
+              it "does not patch the integration multiple times" do
+                expect(FakeIntegration::Patcher).to receive(:patch).and_call_original.once
+
+                instrument
+                instrument
+              end
+            end
+
+            context "when not loaded" do
+              before { allow(FakeIntegration).to receive(:loaded?).and_return(false) }
+
+              it "does not patch the integration" do
+                expect(FakeIntegration::Patcher).to_not receive(:patch)
+
+                instrument
+              end
+            end
+
+            context "when not available" do
+              before { allow(FakeIntegration).to receive(:available?).and_return(false) }
+
+              it "does not patch the integration" do
+                expect(FakeIntegration::Patcher).to_not receive(:patch)
+
+                instrument
+              end
+            end
+
+            context "when not compatible" do
+              before { allow(FakeIntegration).to receive(:compatible?).and_return(false) }
+
+              it "does not patch the integration" do
+                expect(FakeIntegration::Patcher).to_not receive(:patch)
+
+                instrument
+              end
+            end
+
+            context "when not enabled" do
+              # before do
+              #   allow(integration.configuration).to receive(:enabled).and_return(false)
+              # end
```

Comment:
is this intended to be left as a comment?

Reply:
good catch, need to remove